### PR TITLE
feat(editor): Compute node position and connections when creating new nodes in new canvas (no-changelog)

### PR DESCRIPTION
**Pull request by alexgrozav**
## Summary

- Simplified code structure. 
- Moved node creation to `useCanvasOperations`. 
- Added new position calculations.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/N8N-7400/compute-node-position-when-creating-new-nodes
https://linear.app/n8n/issue/N8N-7397/check-if-maximum-node-type-limit-reached

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/views/NodeView.v2.vue [hunk 1]
```diff
@@ -167,23 +192,170 @@ async function initialize() {
...
+		// Load workflow data
+		try {
+			await workflowsStore.fetchWorkflow(workflowIdParam);
+
+			titleChange.titleSet(workflow.value.name, 'IDLE');
+			// @TODO Implement this
+			// await openWorkflow(workflow);
+			// await checkAndInitDebugMode();
+
+			await workflowsStore.initializeEditableWorkflow(workflowIdParam);
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/views/NodeView.v2.vue:267]: Linter warning here: `initializeEditableWorkflow` doesn't return a promise

File: packages/editor-ui/src/stores/workflows.store.ts [hunk 2]
```diff
@@ -77,6 +79,7 @@ import { useProjectsStore } from '@/stores/projects.store';
 import { useSettingsStore } from './settings.store';
 import { useUsersStore } from './users.store';
 import { useTagsStore } from '@/stores/tags.store';
+import { usePostHog } from '@/stores/posthog.store';
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/stores/workflows.store.ts:82]: This import and `useUsersStore` and `useSettingsStore` are not used

File: packages/editor-ui/src/stores/workflows.store.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import {
+	CANVAS_AUTO_ADD_MANUAL_TRIGGER_EXPERIMENT,
 	CHAT_TRIGGER_NODE_TYPE,
 	DEFAULT_NEW_WORKFLOW_NAME,
 	DUPLICATE_POSTFFIX,
 	ERROR_TRIGGER_NODE_TYPE,
 	MAX_WORKFLOW_NAME_LENGTH,
+	NEW_WORKFLOW_ID,
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/stores/workflows.store.ts:8]: Typo? 🤔  None of these newly imported consts are used

**Review by OlegIvaniv — COMMENTED** [review]
Nice job! Left just a few minor comments

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.48.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.48.0)

**Pull request merged**